gpg2: Add page (#3074)
tldr-pages/tldr: pages/common/gpg2.md

# gpg

> GNU Privacy Guard 2.

> > Homepage: <https://docs.releng.linuxfoundation.org/en/latest/gpg.html>.

Reviewer: ```suggestion
> More information: <https://docs.releng.linuxfoundation.org/en/latest/gpg.html>.
```

- List imported keys:

`gpg2 --list-keys`

- Encrypt a specified file for a specified recipient, writing the output to a new file with `.gpg` appended:

`gpg2 --encrypt --recipient {{alice@example.com}} {{path/to/doc.txt}}`

- Encrypt a specified file with only a passphrase, writing the output to a new file with `.gpg` appended:

`gpg2 --symmetric {{path/to/doc.txt}}`

- Decrypt a specified file, writing the result to the standard output:

`gpg2 --decrypt {{path/to/doc.txt.gpg}}`

- Import a public key:

> `gpg2 --import {{path/to/public-key.gpg}}`

Reviewer 2: public_key

> - Export the public key of alice@example.com to the standard output:

Reviewer 2: Here you mention "alice@example.com", but below you write "specified email address". I think we should use just one convention throughout the page. I would go with "specified email address".

Author: Oh yes, I think this was one of the propositions @sbrl suggested that I forgot.

I'll fix it.

# gpg

> GNU Privacy Guard 2.
> Homepage: <https://docs.releng.linuxfoundation.org/en/latest/gpg.html>.

- List imported keys:

`gpg2 --list-keys`

> - Encrypt doc.txt for alice@example.com (output to doc.txt.gpg):

Reviewer 3: Perhaps this will read better?
```suggestion
- Encrypt a specified file with the public key of alice@example.com, writing the output to a new file with `.gpg` appended:
```

Author: Thanks! This has been resolved too.

`gpg2 --encrypt --recipient {{alice@example.com}} {{doc.txt}}`

> - Encrypt doc.txt with only a passphrase (output to doc.txt.gpg):

Reviewer 3: Perhaps this will read better?
```suggestion
- Encrypt a specified file with only a passphrase, writing the output to a new file with `.gpg` appended:
```

Also, could we change `doc.txt` on the line below  to `path/to/doc.txt` please?

Author: Yes, I'll change that.

`gpg2 --symmetric {{doc.txt}}`

> - Decrypt doc.txt.gpg (output to STDOUT):

Reviewer 3: Perhaps this will read better?
```suggestion
- Decrypt a file, writing the result to the standard output:
```

Author: Ditto.

Author: I'd change it to: `Decrypt a specified file, writing the result to the standard output` to maintain consistency with the changes proposed bellow.

What do you think?

Reviewer 3: Sounds great!

> `gpg2 --decrypt {{doc.txt.gpg}}`

Reviewer 3: Perhaps this will read better?
```suggestion
`gpg2 --decrypt {{path/to/doc.txt.gpg}}`
```

Author: Ditto.

- Import a public key:

> `gpg2 --import {{public.gpg}}`

Reviewer 3: Perhaps this will read better?
```suggestion
`gpg2 --import {{path/to/public.gpg}}`
```

Author: Ditto.

> - Export public key for alice@example.com (output to STDOUT):

Reviewer 3: Perhaps this will read better?
```suggestion
- Export the public key of alice@example.com to the standard output:
```

Author: Yes!

`gpg2 --export --armor {{alice@example.com}}`

> - Export private key for alice@example.com (output to STDOUT):

Reviewer 3: Perhaps this will read better?
```suggestion
- Export the private key with a specified email address to the standard output:
```

Author: Ditto.

`gpg2 --export-secret-keys --armor {{alice@example.com}}`

> 

Reviewer 3: Hey! It looks like there's an extra newline at the end of the file that our linter is complaining about. Perhaps you could remove it please?

Author: Yes, I'll do it now !
